### PR TITLE
mixin for form elements

### PR DESCRIPTION
**Pull request by nguasch**
I've created a new mixin for forms elements: inputs, textareas and their own sizes, validations (success or alert) and help text

### CODE REVIEW

File: src/utils/_forms.scss [hunk 1]
```diff
@@ -29,3 +29,84 @@
...
+    $size: medium 40px, small 30px;
+  } @else if($field == 'textarea') {
+    $size: medium 400px, small 200px;
+  }
+
+  @each $type, $h in $size {
+    &-#{$type} {
+      #{$field} {
+        height: $h;
+      }
```
Comment (JoanClaret) [inline src/utils/_forms.scss]: This ouput something like:

```scss
.sui-Form-medium input {
  height: 40px;
}
.sui-Form-small input {
  height: 30px;
}
```
we usually avoid style elements. what do you think about adding a class to the styled element?

File: src/utils/_forms.scss [hunk 2]
```diff
@@ -29,3 +29,84 @@
     display: none;
   }
 }
+
+// Any form field: input, textarea, ...
+// e.g: @include formField(input);
```
Comment (JoanClaret) [inline src/utils/_forms.scss]: a better documentation about how to structure the markup would be much appreciated. ie:
```html
<div class="sui-Form sui-Form-medium sui-Form-success">
  <label>Name</label>
  <input class="sui-Form-input" type="text">
  <span class="sui-Form-success-validation-text">Success message</span>
</div>
```

File: src/utils/_forms.scss [hunk 3]
```diff
@@ -29,3 +29,84 @@
...
+// Help text for form elements
+@mixin help-text {
+  color: $c-gray-light;
+  display: block;
+  font-size: $fz-xs;
+  margin: $m-s 0 0;
+}
+
+// Size for form elements
+@mixin field-size($field) {
```
Comment (JoanClaret) [inline src/utils/_forms.scss]: if i'm not wrong, this should be used like so:
```scss
.sui-Form {
  @include form-field(input);
  @include field-size(input);
}
```
is this correct?

i was wondering why the mixin `field-size` is not called directly from `form-field`.
so we can just:
```scss
.sui-Form {
  @include form-field(input);
}
```

File: src/utils/_forms.scss [hunk 4]
```diff
@@ -29,3 +29,84 @@
     display: none;
   }
 }
+
+// Any form field: input, textarea, ...
+// e.g: @include formField(input);
+
+@mixin form-field($field) {
+  overflow: hidden;
```
Comment (JoanClaret) [inline src/utils/_forms.scss]: i don't understand why does we need this overflow. can you bring some light?

Comment (nguasch) [reply src/utils/_forms.scss]: I think I have _overflowitis_

File: src/utils/_forms.scss (hunk 1 above)
Comment (nguasch) [reply src/utils/_forms.scss]: I agree, but in this case, we can use in different types of elements form, and the name of the class is defined in the react component

File: src/utils/_forms.scss (hunk 2 above)
Comment (nguasch) [reply src/utils/_forms.scss]: Ok.

File: src/utils/_forms.scss (hunk 3 above)
Comment (nguasch) [reply src/utils/_forms.scss]: I think it's better split to use by separate

Comment (nguasch) [reply src/utils/_forms.scss]: not really 😓 
I've moved it and now it's called from `form-field`

File: src/utils/_forms.scss [hunk 5]
```diff
@@ -29,3 +29,93 @@
     display: none;
   }
 }
+
+//.sui-Form {
+//  @include form-field(input);
+//}
+
+//<div class="sui-AtomInput sui-AtomInput-success">
```
Comment (miduga) [inline src/utils/_forms.scss]: Classnames are not following SUIT CSS as we're using in Schibsted. https://suitcss.github.io/

`sui-AtomInput sui-AtomInput--success`
`sui-AtomInput-inputWrapper`
`sui-AtomInput--medium`

**Pull request merged**